### PR TITLE
Transform Text function exception into runtime error

### PR DESCRIPTION
**Pull request by gregli-msft**
If the text formatter used in the Text function ran into a format it did not understand, for example a simple "i", then the code would throw and the exception would not be caught.

The diff is broken up and looks more involved then it is, all that was done is to add a try/catch with the appropriate addition of tabs for the extra block.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 1]
```diff
@@ -266,4 +266,8 @@ Error({Kind:ErrorKind.BadLanguageCode})
 "domingo enero 31"
 
 >> Text(DateTimeValue("12:34:56"),"yyyy-mm-dd hh:mm:ss")
-"1899-12-30 12:34:56"
\ No newline at end of file
+"1899-12-30 12:34:56"
+
+// This would previously throw an exception in the interpreter by the C# text formatter on the "i"
+>> Text( 12, "i" )
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:272]: Should this be an error at all?

image

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**